src/http/server/handlers/static_file_handler.cr

```diff
@@ -64,6 +64,14 @@ class HTTP::StaticFileHandler
       context.response.content_type = "text/html"
       directory_listing(context.response, request_path, file_path)
     elsif is_file
+      last_modified = File.stat(file_path).mtime.to_s("%a, %-d %h %C%y %T GMT")
+      if if_modified_since = context.request.headers.fetch("If-Modified-Since", nil)
+        if last_modified == if_modified_since
```

Comment:
According to [RFC 7232](https://tools.ietf.org/html/rfc7232#section-3.3) the server should return 304 if the "last modification date is **earlier than or equal**" to the date provided in the" `If-Modified-Since` header.
So you'd need to parse the field-value and compare the times with `<=`. It is also necessary to parse the string because the client might not send the value of the `Last-Modified` header verbatim but with a different format.
 
Unfortunately, there is currently no parser in stdlib for that format but with #4729 there would be `Time.parse_rfc2822` which would at least be sufficient to parse the recommended format. Until that PR is merged, you might just use `Time::Format.new("%a, %d %b %Y %H:%M:%S %z")` directly and add a `FIXME` note do address broader compatibility later.